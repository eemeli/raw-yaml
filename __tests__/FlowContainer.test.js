import { commonTests, testParse } from './common'

describe('parse simple array', () => {
  for (const name in commonTests) {
    const props = Object.assign({ str: '[ "one", 2, three four ]' }, commonTests[name])
    test(name, () => testParse(props))
  }
})

describe('parse deep array', () => {
  const str = `[ "one", 2,
    three four,
[five, \t"6,"] ]`
  const itemTest = (node) => expect(node.items[7].items[3].rawValue).toBe('"6,"')
  for (const name in commonTests) {
    const props = Object.assign({ str, test: itemTest }, commonTests[name])
    if (props.comment) props.test = (node) => {
      expect(node.items.some(n => n.comment === props.comment))
      expect(node.items[8].items[3].rawValue).toBe('"6,"')
    }
    test(name, () => testParse(props))
  }
})

describe('parse simple object', () => {
  for (const name in commonTests) {
    const props = Object.assign({ str: '{ "first":"one", 2, three: four,?fi:ve }' }, commonTests[name])
    test(name, () => testParse(props))
  }
})
