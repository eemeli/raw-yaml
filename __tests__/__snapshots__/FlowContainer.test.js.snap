// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse deep array bare 1`] = `
FlowContainer {
  "anchor": null,
  "commentRange": null,
  "items": Array [
    "[",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 7,
        "start": 2,
      },
      "raw": "\\"one\\"",
      "tag": null,
      "type": "DOUBLE",
      "valueRange": Range {
        "end": 7,
        "start": 2,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 10,
        "start": 9,
      },
      "raw": "2",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 10,
        "start": 9,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 26,
        "start": 16,
      },
      "raw": "three four",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 26,
        "start": 16,
      },
    },
    ",",
    FlowContainer {
      "anchor": null,
      "commentRange": null,
      "items": Array [
        "[",
        Scalar {
          "anchor": null,
          "blockStyle": null,
          "commentRange": null,
          "range": Range {
            "end": 33,
            "start": 29,
          },
          "raw": "five",
          "tag": null,
          "type": "PLAIN",
          "valueRange": Range {
            "end": 33,
            "start": 29,
          },
        },
        ",",
        Scalar {
          "anchor": null,
          "blockStyle": null,
          "commentRange": null,
          "range": Range {
            "end": 40,
            "start": 36,
          },
          "raw": "\\"6,\\"",
          "tag": null,
          "type": "DOUBLE",
          "valueRange": Range {
            "end": 40,
            "start": 36,
          },
        },
        "]",
      ],
      "range": Range {
        "end": 42,
        "start": 28,
      },
      "tag": null,
      "type": "FLOW_SEQ",
      "valueRange": Range {
        "end": 41,
        "start": 28,
      },
    },
    "]",
  ],
  "range": Range {
    "end": 43,
    "start": 0,
  },
  "tag": null,
  "type": "FLOW_SEQ",
  "valueRange": Range {
    "end": 43,
    "start": 0,
  },
}
`;

exports[`parse deep array complex mapping key 1`] = `
FlowContainer {
  "anchor": null,
  "commentRange": null,
  "items": Array [
    "[",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 9,
        "start": 4,
      },
      "raw": "\\"one\\"",
      "tag": null,
      "type": "DOUBLE",
      "valueRange": Range {
        "end": 9,
        "start": 4,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 12,
        "start": 11,
      },
      "raw": "2",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 12,
        "start": 11,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 28,
        "start": 18,
      },
      "raw": "three four",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 28,
        "start": 18,
      },
    },
    ",",
    FlowContainer {
      "anchor": null,
      "commentRange": null,
      "items": Array [
        "[",
        Scalar {
          "anchor": null,
          "blockStyle": null,
          "commentRange": null,
          "range": Range {
            "end": 35,
            "start": 31,
          },
          "raw": "five",
          "tag": null,
          "type": "PLAIN",
          "valueRange": Range {
            "end": 35,
            "start": 31,
          },
        },
        ",",
        Scalar {
          "anchor": null,
          "blockStyle": null,
          "commentRange": null,
          "range": Range {
            "end": 42,
            "start": 38,
          },
          "raw": "\\"6,\\"",
          "tag": null,
          "type": "DOUBLE",
          "valueRange": Range {
            "end": 42,
            "start": 38,
          },
        },
        "]",
      ],
      "range": Range {
        "end": 44,
        "start": 30,
      },
      "tag": null,
      "type": "FLOW_SEQ",
      "valueRange": Range {
        "end": 43,
        "start": 30,
      },
    },
    "]",
  ],
  "range": Range {
    "end": 46,
    "start": 2,
  },
  "tag": null,
  "type": "FLOW_SEQ",
  "valueRange": Range {
    "end": 45,
    "start": 2,
  },
}
`;

exports[`parse deep array flow seq value 1`] = `
FlowContainer {
  "anchor": null,
  "commentRange": null,
  "items": Array [
    "[",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 9,
        "start": 4,
      },
      "raw": "\\"one\\"",
      "tag": null,
      "type": "DOUBLE",
      "valueRange": Range {
        "end": 9,
        "start": 4,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 12,
        "start": 11,
      },
      "raw": "2",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 12,
        "start": 11,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 28,
        "start": 18,
      },
      "raw": "three four",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 28,
        "start": 18,
      },
    },
    ",",
    FlowContainer {
      "anchor": null,
      "commentRange": null,
      "items": Array [
        "[",
        Scalar {
          "anchor": null,
          "blockStyle": null,
          "commentRange": null,
          "range": Range {
            "end": 35,
            "start": 31,
          },
          "raw": "five",
          "tag": null,
          "type": "PLAIN",
          "valueRange": Range {
            "end": 35,
            "start": 31,
          },
        },
        ",",
        Scalar {
          "anchor": null,
          "blockStyle": null,
          "commentRange": null,
          "range": Range {
            "end": 42,
            "start": 38,
          },
          "raw": "\\"6,\\"",
          "tag": null,
          "type": "DOUBLE",
          "valueRange": Range {
            "end": 42,
            "start": 38,
          },
        },
        "]",
      ],
      "range": Range {
        "end": 44,
        "start": 30,
      },
      "tag": null,
      "type": "FLOW_SEQ",
      "valueRange": Range {
        "end": 43,
        "start": 30,
      },
    },
    "]",
  ],
  "range": Range {
    "end": 46,
    "start": 2,
  },
  "tag": null,
  "type": "FLOW_SEQ",
  "valueRange": Range {
    "end": 45,
    "start": 2,
  },
}
`;

exports[`parse deep array indented block 1`] = `
FlowContainer {
  "anchor": null,
  "commentRange": null,
  "items": Array [
    "[",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 13,
        "start": 8,
      },
      "raw": "\\"one\\"",
      "tag": null,
      "type": "DOUBLE",
      "valueRange": Range {
        "end": 13,
        "start": 8,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 16,
        "start": 15,
      },
      "raw": "2",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 16,
        "start": 15,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 32,
        "start": 22,
      },
      "raw": "three four",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 32,
        "start": 22,
      },
    },
    ",",
    FlowContainer {
      "anchor": null,
      "commentRange": null,
      "items": Array [
        "[",
        Scalar {
          "anchor": null,
          "blockStyle": null,
          "commentRange": null,
          "range": Range {
            "end": 39,
            "start": 35,
          },
          "raw": "five",
          "tag": null,
          "type": "PLAIN",
          "valueRange": Range {
            "end": 39,
            "start": 35,
          },
        },
        ",",
        Scalar {
          "anchor": null,
          "blockStyle": null,
          "commentRange": null,
          "range": Range {
            "end": 46,
            "start": 42,
          },
          "raw": "\\"6,\\"",
          "tag": null,
          "type": "DOUBLE",
          "valueRange": Range {
            "end": 46,
            "start": 42,
          },
        },
        "]",
      ],
      "range": Range {
        "end": 48,
        "start": 34,
      },
      "tag": null,
      "type": "FLOW_SEQ",
      "valueRange": Range {
        "end": 47,
        "start": 34,
      },
    },
    "]",
  ],
  "range": Range {
    "end": 49,
    "start": 6,
  },
  "tag": null,
  "type": "FLOW_SEQ",
  "valueRange": Range {
    "end": 49,
    "start": 6,
  },
}
`;

exports[`parse deep array newline before & after 1`] = `
FlowContainer {
  "anchor": null,
  "commentRange": null,
  "items": Array [
    "[",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 8,
        "start": 3,
      },
      "raw": "\\"one\\"",
      "tag": null,
      "type": "DOUBLE",
      "valueRange": Range {
        "end": 8,
        "start": 3,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 11,
        "start": 10,
      },
      "raw": "2",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 11,
        "start": 10,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 27,
        "start": 17,
      },
      "raw": "three four",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 27,
        "start": 17,
      },
    },
    ",",
    FlowContainer {
      "anchor": null,
      "commentRange": null,
      "items": Array [
        "[",
        Scalar {
          "anchor": null,
          "blockStyle": null,
          "commentRange": null,
          "range": Range {
            "end": 34,
            "start": 30,
          },
          "raw": "five",
          "tag": null,
          "type": "PLAIN",
          "valueRange": Range {
            "end": 34,
            "start": 30,
          },
        },
        ",",
        Scalar {
          "anchor": null,
          "blockStyle": null,
          "commentRange": null,
          "range": Range {
            "end": 41,
            "start": 37,
          },
          "raw": "\\"6,\\"",
          "tag": null,
          "type": "DOUBLE",
          "valueRange": Range {
            "end": 41,
            "start": 37,
          },
        },
        "]",
      ],
      "range": Range {
        "end": 43,
        "start": 29,
      },
      "tag": null,
      "type": "FLOW_SEQ",
      "valueRange": Range {
        "end": 42,
        "start": 29,
      },
    },
    "]",
  ],
  "range": Range {
    "end": 44,
    "start": 1,
  },
  "tag": null,
  "type": "FLOW_SEQ",
  "valueRange": Range {
    "end": 44,
    "start": 1,
  },
}
`;

exports[`parse deep array seq value 1`] = `
FlowContainer {
  "anchor": null,
  "commentRange": null,
  "items": Array [
    "[",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 9,
        "start": 4,
      },
      "raw": "\\"one\\"",
      "tag": null,
      "type": "DOUBLE",
      "valueRange": Range {
        "end": 9,
        "start": 4,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 12,
        "start": 11,
      },
      "raw": "2",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 12,
        "start": 11,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 28,
        "start": 18,
      },
      "raw": "three four",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 28,
        "start": 18,
      },
    },
    ",",
    FlowContainer {
      "anchor": null,
      "commentRange": null,
      "items": Array [
        "[",
        Scalar {
          "anchor": null,
          "blockStyle": null,
          "commentRange": null,
          "range": Range {
            "end": 35,
            "start": 31,
          },
          "raw": "five",
          "tag": null,
          "type": "PLAIN",
          "valueRange": Range {
            "end": 35,
            "start": 31,
          },
        },
        ",",
        Scalar {
          "anchor": null,
          "blockStyle": null,
          "commentRange": null,
          "range": Range {
            "end": 42,
            "start": 38,
          },
          "raw": "\\"6,\\"",
          "tag": null,
          "type": "DOUBLE",
          "valueRange": Range {
            "end": 42,
            "start": 38,
          },
        },
        "]",
      ],
      "range": Range {
        "end": 44,
        "start": 30,
      },
      "tag": null,
      "type": "FLOW_SEQ",
      "valueRange": Range {
        "end": 43,
        "start": 30,
      },
    },
    "]",
  ],
  "range": Range {
    "end": 45,
    "start": 2,
  },
  "tag": null,
  "type": "FLOW_SEQ",
  "valueRange": Range {
    "end": 45,
    "start": 2,
  },
}
`;

exports[`parse deep array with comment 1`] = `
FlowContainer {
  "anchor": null,
  "commentRange": null,
  "items": Array [
    "[",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 10,
        "start": 5,
      },
      "raw": "\\"one\\"",
      "tag": null,
      "type": "DOUBLE",
      "valueRange": Range {
        "end": 10,
        "start": 5,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 13,
        "start": 12,
      },
      "raw": "2",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 13,
        "start": 12,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": Range {
        "end": 31,
        "start": 16,
      },
      "range": null,
      "raw": "",
      "tag": null,
      "type": "COMMENT",
      "valueRange": Range {
        "end": 15,
        "start": 15,
      },
    },
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 46,
        "start": 36,
      },
      "raw": "three four",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 46,
        "start": 36,
      },
    },
    ",",
    FlowContainer {
      "anchor": null,
      "commentRange": null,
      "items": Array [
        "[",
        Scalar {
          "anchor": null,
          "blockStyle": null,
          "commentRange": null,
          "range": Range {
            "end": 53,
            "start": 49,
          },
          "raw": "five",
          "tag": null,
          "type": "PLAIN",
          "valueRange": Range {
            "end": 53,
            "start": 49,
          },
        },
        ",",
        Scalar {
          "anchor": null,
          "blockStyle": null,
          "commentRange": null,
          "range": Range {
            "end": 60,
            "start": 56,
          },
          "raw": "\\"6,\\"",
          "tag": null,
          "type": "DOUBLE",
          "valueRange": Range {
            "end": 60,
            "start": 56,
          },
        },
        "]",
      ],
      "range": Range {
        "end": 62,
        "start": 48,
      },
      "tag": null,
      "type": "FLOW_SEQ",
      "valueRange": Range {
        "end": 61,
        "start": 48,
      },
    },
    "]",
  ],
  "range": Range {
    "end": 63,
    "start": 3,
  },
  "tag": null,
  "type": "FLOW_SEQ",
  "valueRange": Range {
    "end": 63,
    "start": 3,
  },
}
`;

exports[`parse deep array with props 1`] = `
FlowContainer {
  "anchor": "anchor",
  "commentRange": null,
  "items": Array [
    "[",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 23,
        "start": 18,
      },
      "raw": "\\"one\\"",
      "tag": null,
      "type": "DOUBLE",
      "valueRange": Range {
        "end": 23,
        "start": 18,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 26,
        "start": 25,
      },
      "raw": "2",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 26,
        "start": 25,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 42,
        "start": 32,
      },
      "raw": "three four",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 42,
        "start": 32,
      },
    },
    ",",
    FlowContainer {
      "anchor": null,
      "commentRange": null,
      "items": Array [
        "[",
        Scalar {
          "anchor": null,
          "blockStyle": null,
          "commentRange": null,
          "range": Range {
            "end": 49,
            "start": 45,
          },
          "raw": "five",
          "tag": null,
          "type": "PLAIN",
          "valueRange": Range {
            "end": 49,
            "start": 45,
          },
        },
        ",",
        Scalar {
          "anchor": null,
          "blockStyle": null,
          "commentRange": null,
          "range": Range {
            "end": 56,
            "start": 52,
          },
          "raw": "\\"6,\\"",
          "tag": null,
          "type": "DOUBLE",
          "valueRange": Range {
            "end": 56,
            "start": 52,
          },
        },
        "]",
      ],
      "range": Range {
        "end": 58,
        "start": 44,
      },
      "tag": null,
      "type": "FLOW_SEQ",
      "valueRange": Range {
        "end": 57,
        "start": 44,
      },
    },
    "]",
  ],
  "range": Range {
    "end": 59,
    "start": 2,
  },
  "tag": "tag!",
  "type": "FLOW_SEQ",
  "valueRange": Range {
    "end": 59,
    "start": 16,
  },
}
`;

exports[`parse simple array bare 1`] = `
FlowContainer {
  "anchor": null,
  "commentRange": null,
  "items": Array [
    "[",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 7,
        "start": 2,
      },
      "raw": "\\"one\\"",
      "tag": null,
      "type": "DOUBLE",
      "valueRange": Range {
        "end": 7,
        "start": 2,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 10,
        "start": 9,
      },
      "raw": "2",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 10,
        "start": 9,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 23,
        "start": 12,
      },
      "raw": "three four",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 22,
        "start": 12,
      },
    },
    "]",
  ],
  "range": Range {
    "end": 24,
    "start": 0,
  },
  "tag": null,
  "type": "FLOW_SEQ",
  "valueRange": Range {
    "end": 24,
    "start": 0,
  },
}
`;

exports[`parse simple array complex mapping key 1`] = `
FlowContainer {
  "anchor": null,
  "commentRange": null,
  "items": Array [
    "[",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 9,
        "start": 4,
      },
      "raw": "\\"one\\"",
      "tag": null,
      "type": "DOUBLE",
      "valueRange": Range {
        "end": 9,
        "start": 4,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 12,
        "start": 11,
      },
      "raw": "2",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 12,
        "start": 11,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 25,
        "start": 14,
      },
      "raw": "three four",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 24,
        "start": 14,
      },
    },
    "]",
  ],
  "range": Range {
    "end": 27,
    "start": 2,
  },
  "tag": null,
  "type": "FLOW_SEQ",
  "valueRange": Range {
    "end": 26,
    "start": 2,
  },
}
`;

exports[`parse simple array flow seq value 1`] = `
FlowContainer {
  "anchor": null,
  "commentRange": null,
  "items": Array [
    "[",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 9,
        "start": 4,
      },
      "raw": "\\"one\\"",
      "tag": null,
      "type": "DOUBLE",
      "valueRange": Range {
        "end": 9,
        "start": 4,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 12,
        "start": 11,
      },
      "raw": "2",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 12,
        "start": 11,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 25,
        "start": 14,
      },
      "raw": "three four",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 24,
        "start": 14,
      },
    },
    "]",
  ],
  "range": Range {
    "end": 27,
    "start": 2,
  },
  "tag": null,
  "type": "FLOW_SEQ",
  "valueRange": Range {
    "end": 26,
    "start": 2,
  },
}
`;

exports[`parse simple array indented block 1`] = `
FlowContainer {
  "anchor": null,
  "commentRange": null,
  "items": Array [
    "[",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 13,
        "start": 8,
      },
      "raw": "\\"one\\"",
      "tag": null,
      "type": "DOUBLE",
      "valueRange": Range {
        "end": 13,
        "start": 8,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 16,
        "start": 15,
      },
      "raw": "2",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 16,
        "start": 15,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 29,
        "start": 18,
      },
      "raw": "three four",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 28,
        "start": 18,
      },
    },
    "]",
  ],
  "range": Range {
    "end": 30,
    "start": 6,
  },
  "tag": null,
  "type": "FLOW_SEQ",
  "valueRange": Range {
    "end": 30,
    "start": 6,
  },
}
`;

exports[`parse simple array newline before & after 1`] = `
FlowContainer {
  "anchor": null,
  "commentRange": null,
  "items": Array [
    "[",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 8,
        "start": 3,
      },
      "raw": "\\"one\\"",
      "tag": null,
      "type": "DOUBLE",
      "valueRange": Range {
        "end": 8,
        "start": 3,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 11,
        "start": 10,
      },
      "raw": "2",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 11,
        "start": 10,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 24,
        "start": 13,
      },
      "raw": "three four",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 23,
        "start": 13,
      },
    },
    "]",
  ],
  "range": Range {
    "end": 25,
    "start": 1,
  },
  "tag": null,
  "type": "FLOW_SEQ",
  "valueRange": Range {
    "end": 25,
    "start": 1,
  },
}
`;

exports[`parse simple array seq value 1`] = `
FlowContainer {
  "anchor": null,
  "commentRange": null,
  "items": Array [
    "[",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 9,
        "start": 4,
      },
      "raw": "\\"one\\"",
      "tag": null,
      "type": "DOUBLE",
      "valueRange": Range {
        "end": 9,
        "start": 4,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 12,
        "start": 11,
      },
      "raw": "2",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 12,
        "start": 11,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 25,
        "start": 14,
      },
      "raw": "three four",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 24,
        "start": 14,
      },
    },
    "]",
  ],
  "range": Range {
    "end": 26,
    "start": 2,
  },
  "tag": null,
  "type": "FLOW_SEQ",
  "valueRange": Range {
    "end": 26,
    "start": 2,
  },
}
`;

exports[`parse simple array with comment 1`] = `
FlowContainer {
  "anchor": null,
  "commentRange": Range {
    "end": 44,
    "start": 29,
  },
  "items": Array [
    "[",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 10,
        "start": 5,
      },
      "raw": "\\"one\\"",
      "tag": null,
      "type": "DOUBLE",
      "valueRange": Range {
        "end": 10,
        "start": 5,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 13,
        "start": 12,
      },
      "raw": "2",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 13,
        "start": 12,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 26,
        "start": 15,
      },
      "raw": "three four",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 25,
        "start": 15,
      },
    },
    "]",
  ],
  "range": Range {
    "end": 44,
    "start": 3,
  },
  "tag": null,
  "type": "FLOW_SEQ",
  "valueRange": Range {
    "end": 27,
    "start": 3,
  },
}
`;

exports[`parse simple array with props 1`] = `
FlowContainer {
  "anchor": "anchor",
  "commentRange": null,
  "items": Array [
    "[",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 23,
        "start": 18,
      },
      "raw": "\\"one\\"",
      "tag": null,
      "type": "DOUBLE",
      "valueRange": Range {
        "end": 23,
        "start": 18,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 26,
        "start": 25,
      },
      "raw": "2",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 26,
        "start": 25,
      },
    },
    ",",
    Scalar {
      "anchor": null,
      "blockStyle": null,
      "commentRange": null,
      "range": Range {
        "end": 39,
        "start": 28,
      },
      "raw": "three four",
      "tag": null,
      "type": "PLAIN",
      "valueRange": Range {
        "end": 38,
        "start": 28,
      },
    },
    "]",
  ],
  "range": Range {
    "end": 40,
    "start": 2,
  },
  "tag": "tag!",
  "type": "FLOW_SEQ",
  "valueRange": Range {
    "end": 40,
    "start": 16,
  },
}
`;
